services/parser-database/apiSpec.yaml

```diff
@@ -0,0 +1,130 @@
+swagger: "2.0"
+info:
+  description: "API description for the Parser/Database component of the Road Search Engine."
+  version: "1.0.0"
+  title: "Parser/Database Component"
+  contact:
+    email: "jaimehisao@google.com"
+host: "localhost"
+basePath: "/v1"
+schemes:
+- "https"
+- "http"
+paths:
+  /parse:
+    post:
+      tags:
+      - "parser"
+      summary: "Trigger a document download, parse, and metadata tagging"
+      responses:
+        "200":
+         description: "Document was parsed successfully"
+        "409":
+          description: "Parser service is already running."
+        "500":
+          description: "Server cannot parse the document and add it to the database. Likely if the document is not in the specified URL or if the database is offline."
+  /articles/byKeyword:
+    post:
+      tags:
+      - "articles"
+      summary: "Finds any article that contains the specified keyword(s)"
+      description: ""
+      operationId: "returnArticleIDs"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "Keywords that need to be matched with an article in the database"
+        required: true
+        schema:
+          $ref: '#/definitions/ArticleKeywordsQuery'
+      responses:
+        "400":
+          description: "No articles matched keyword query"
+        "200":
+          description: "Successful Operation"
+          schema:
+            $ref: '#/definitions/ArticleKeywords'
+  /articles/{id}:
+    get:
+      tags:
+      - "articles"
+      summary: "Returns the text of the requested article"
+      description: "Single article ID value"
+      operationId: "findTextByID"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "id"
+        in: "path"
+        required: true
+        description: "ID of the requested article"
+        type: "string"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Article"
+        "400":
+          description: "Invalid Article ID"
+        "500":
+          description: "Error when retrieving the Article from the database"
+definitions:
+  Article:
+    type: "object"
+    properties:
+      articleNumber:
+        type: "integer"
+        format: "int 64"
+      text:
+        type: "string"
+      timeToRead:
+        type: "integer"
+        format: "int 64"
+      pageNumber:
+        type: "integer"
+        format: "int 64"
```

Comment:
By default, the properties are considered optional if they are not marked as `required`. IIUC, all these properties must be present, so they should be marked `required`